duckduckgo/macos-browser
# Updates sorting for Autofill UI Logins

Task/Issue URL: https://app.asana.com/0/1204099484721401/1205016455499671/f

**Description**:
The sorting algo for Logins accounts prioritizes the last update date in order to show potentially better suggestions to users on Autofill, but as this is also used to select the best possible option from the Autofill UI, the last updated item will be selected by default, if you don't have saved logins for a site.

This fixes the issue, so if there's not an exact Match or TLD match for the visited domain, the first item is selected


**Steps to test this PR**:
1. Make sure you have a few logins in your vault, added across several days
2. Using the PROD version, visit a site for which you don't have a login stored
3. Open up the Autofill UI, and observe a 'random' item (last updated) item is selected
4. Run this PR and do the same steps
5. Observe the first item in the list is now selected



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
